File: lib/src/rust/api/ncm_decrypt.dart
// NCM 解密 FFI 绑定
// 注意：此文件需要通过 flutter_rust_bridge_codegen generate 重新生成
// 当前为手动编写的临时版本，功能完整后请运行代码生成器替换

import '../frb_generated.dart';
import 'package:flutter_rust_bridge/flutter_rust_bridge_for_generated.dart';

String ncmScanFilesJson({required String dir}) =>
    RustLib.instance.api.crateApiNcmDecryptNcmScanFilesJson(dir: dir);

String ncmGetProgressJson() =>
    RustLib.instance.api.crateApiNcmDecryptNcmGetProgressJson();

String ncmGetResultJson() =>
    RustLib.instance.api.crateApiNcmDecryptNcmGetResultJson();

void ncmDecryptStart({required String configJson}) =>
    RustLib.instance.api.crateApiNcmDecryptNcmDecryptStart(configJson: configJson);

void ncmDecryptCancel() =>
    RustLib.instance.api.crateApiNcmDecryptNcmDecryptCancel();

File: rust/src/api/ncm_decrypt.rs
use flutter_rust_bridge::frb;

#[frb(sync)]
pub fn ncm_scan_files_json(dir: String) -> String {
    ncm_decrypt::ncm_scan_files_json(dir)
}

#[frb(sync)]
pub fn ncm_get_progress_json() -> String {
    ncm_decrypt::ncm_get_progress_json()
}

#[frb(sync)]
pub fn ncm_get_result_json() -> String {
    ncm_decrypt::ncm_get_result_json()
}

#[frb(sync)]
pub fn ncm_decrypt_start(config_json: String) {
    ncm_decrypt::ncm_decrypt_start(config_json);
}

#[frb(sync)]
pub fn ncm_decrypt_cancel() {
    ncm_decrypt::ncm_decrypt_cancel();
}
